internal/bzlmod/go_deps.bzl

```diff
@@ -374,6 +377,37 @@ def _go_deps_impl(module_ctx):
         }.keys(),
     )
 
+def _create_go_repository(path, module, gazelle_overrides, module_overrides, sums):
+    go_repository(
+        name=module.repo_name,
+        importpath=path,
+        sum=_get_sum_from_module(path, module, sums),
+        replace=getattr(module, "replace", None),
```

Comment:
We should fail if a user specifies both a `module_override` and an `archive_override` for the same dep.

Reply:
Sounds good, updated by duplicating the a calls to `_fail_on_duplicate_overrides` and adding a comment.

Let me know if you think this would be better checked after iteration.